tests/sentry/integrations/github/test_client.py

```diff
@@ -1297,6 +1302,78 @@ def test_get_blame_for_files_full_response(self, get_jwt):
+    @responses.activate
+    def test_get_cached_blame_for_files_full_response(self, get_jwt):
+        """
+        Tests that the cached commits are returned with full response
+        """
+        responses.add(
+            method=responses.POST,
+            url="https://api.github.com/graphql",
+            json={
+                "data": self.data,
+            },
+            content_type="application/json",
+        )
+
+        data = create_blame_query(
+            generate_file_path_mapping([self.file1, self.file2, self.file3]), extra={}
+        )
+        cache_key = self.github_client.get_cache_key("/graphql", data)
+        assert self.github_client.check_cache(cache_key) is None
+        response = self.github_client.get_blame_for_files(
+            [self.file1, self.file2, self.file3], extra={}
+        )
+
+        self.assertEqual(
+            response,
+            [
+                FileBlameInfo(
+                    **asdict(self.file1),
+                    commit=CommitInfo(
+                        commitId="123",
+                        commitAuthorName="foo1",
+                        commitAuthorEmail="foo1@example.com",
+                        commitMessage="hello",
+                        committedDate=datetime(2022, 1, 1, 0, 0, 0, tzinfo=timezone.utc),
+                    ),
+                ),
+                FileBlameInfo(
+                    **asdict(self.file2),
+                    commit=CommitInfo(
+                        commitId="456",
+                        commitAuthorName="foo2",
+                        commitAuthorEmail="foo2@example.com",
+                        commitMessage="hello",
+                        committedDate=datetime(2021, 1, 1, 0, 0, 0, tzinfo=timezone.utc),
+                    ),
+                ),
+                FileBlameInfo(
+                    **asdict(self.file3),
+                    commit=CommitInfo(
+                        commitId="789",
+                        commitAuthorName="foo3",
+                        commitAuthorEmail="foo3@example.com",
+                        commitMessage="hello",
+                        committedDate=datetime(2020, 1, 1, 0, 0, 0, tzinfo=timezone.utc),
+                    ),
+                ),
+            ],
+        )
+        assert self.github_client.check_cache(cache_key)["data"] == self.data
+        # Calling a second time should work
+        response = self.github_client.get_blame_for_files(
+            [self.file1, self.file2, self.file3], extra={}
+        )
+        assert self.github_client.check_cache(cache_key)["data"] == self.data
+        # Calling again after the cache has been cleared should still work
+        cache.delete(cache_key)
+        response = self.github_client.get_blame_for_files(
+            [self.file1, self.file2, self.file3], extra={}
+        )
+        assert self.github_client.check_cache(cache_key)["data"] == self.data
```

Comment:
Would be helpful to add an assertion here that ensures different calls don't return the same thing, like:

```python
assert self.github_client.get_blame_for_files([self.file1, self.file2]) != response
```